src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,273 @@
+                        "href",
+                        systemConfig.getDataverseSiteUrl() + "/api/datasets/export?exporter=schema.org&persistentId=" + ds.getProtocol() + ":" + ds.getAuthority() + "/" + ds.getIdentifier()
+                ).add(
+                        "type",
+                        "application/json+ld"
+                )
+        );
+        JsonArrayBuilder linkset = Json.createArrayBuilder();
+        JsonObjectBuilder mandatory = jsonObjectBuilder()
+                .add("anchor", landingPage)
+                .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                .add("type", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage")))
+                .add("author", authors)
+                .add("license", jsonObjectBuilder().add("href", license))
+                .add("item", items).add("describedby", mediaTypes);
+        linkset.add(mandatory);
+
+        if (useDefaultFileType) {
+            for (FileMetadata fm:fms){
+                DataFile df = fm.getDataFile();
+                JsonObjectBuilder itemAnchor = jsonObjectBuilder().add("anchor", getPublicDownloadUrl(df));
+                itemAnchor.add("collection", Json.createArrayBuilder().add(jsonObjectBuilder()
+                        .add("href", landingPage)).add(jsonObjectBuilder().add("type", defaultFileTypeValue)));
+                linkset.add(itemAnchor);
+            }
+        } else {
+            for (FileMetadata fm:fms){
+                DataFile df = fm.getDataFile();
+                JsonObjectBuilder itemAnchor = jsonObjectBuilder().add("anchor", getPublicDownloadUrl(df));
+                itemAnchor.add("collection", Json.createArrayBuilder().add(jsonObjectBuilder()
+                        .add("href", landingPage)));
+                linkset.add(itemAnchor);
+            }
+        }
+
+        return linkset;
+    }
+
+
+    private String getPublicDownloadUrl(DataFile dataFile) {
+        try {
+            StorageIO<DataFile> storageIO = dataFile.getStorageIO();
+            if (storageIO instanceof SwiftAccessIO) {
+                String fileDownloadUrl = null;
+                try {
+                    SwiftAccessIO<DataFile> swiftIO = (SwiftAccessIO<DataFile>) storageIO;
+                    swiftIO.open();
+                    //if its a public install, lets just give users the permanent URL!
+                    if (systemConfig.isPublicInstall()){
+                        fileDownloadUrl = swiftIO.getRemoteUrl();
+                    } else {
+                        //TODO: if a user has access to this file, they should be given the swift url
+                        // perhaps even we could use this as the "private url"
+                        fileDownloadUrl = swiftIO.getTemporarySwiftUrl();
+                    }
+                    return fileDownloadUrl;
+
+                } catch (Exception e) {
```

Comment:
* Don't catch `Exception`. It includes a whole lot of types that you do **not** want to catch, for example NullPointerException. Always catch a specific subtype of Exception.
* Don't just print stacktraces to the STDERR. Dataverse has a log file.